### ISSUE TITLE
Have UserNotificationCondition and RemoteNotificationCondition been deprecated?

### ISSUE THREAD

**Issue by Utsira** [ProcedureKit/ProcedureKit#719]
I'm migrating a project from Swift 2.2 to 3. In the Operations framework, it made use of UserNotificationCondition and RemoteNotificationCondition. These aren't currently present in ProcedureKit, are there any plans to reintroduce them?

Are there any tips on migrating custom conditions from Operations to ProcedureKit?

**Comment by swiftlyfalling** [comment]
@Utsira: You are correct, those Conditions haven't made the cut. @danthorpe might have some more ideas as to why and whether they will be reintroduced.

As for migrating custom conditions from Operations 3.x to ProcedureKit 4.x, here's a quick review of some of the key changes: https://github.com/ProcedureKit/ProcedureKit/pull/681 (Look under "_Breaking Changes (from prior ProcedureKit v4 betas)_")

**Comment by danthorpe** [comment]
Hi @Utsira - yeah, unfortunately, these conditions didn't quite make it to _ProcedureKit_ or rather (_ProcedureKitMobile_). Partly this was just a perceived lack of priority on my part (I got the sense that they are rarely used), and also, I didn't really like how they were implemented from a testability point of view.

However, I'm not adverse to the functionality coming back, its just that we don't have the time to work on it at the moment. But pull requests are welcome!!

Also, to add to the above, @swiftlyfalling has written about migration in the wiki: [Migrating from Operations](https://github.com/ProcedureKit/ProcedureKit/wiki/Migrating-from%3A-%22Operations%22-%28ProcedureKit%29-3.x).

**Comment by danthorpe** [comment]
@Utsira for your own project however, it should be relatively straightforward to include the `UserNotificationCondition` etc from _Operations_ in your own project and migrate it as a _ProcedureKit_ condition. This could even form the basis of a pull-request.
